homeassistant/components/evohome/__init__.py

```diff
@@ -118,91 +113,159 @@
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Create a (EMEA/EU-based) Honeywell TCC system."""
+class EvoSession:
+    """Class for evohome client instantiation & authentication."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the evohome broker and its data structure."""
+
+        self.hass = hass
+
+        self._session = async_get_clientsession(hass)
+        self._store = Store[dict[str, Any]](hass, STORAGE_VER, STORAGE_KEY)
+
+        # the main client, which uses the newer API
+        self.client_v2: evo.EvohomeClient | None = None
+        self._tokens: dict[str, Any] = {}
+
+        # the older client can be used to obtain high-precision temps (only)
+        self.client_v1: ev1.EvohomeClient | None = None
+        self.session_id: str | None = None
+
+    async def authenticate(self, username: str, password: str) -> None:
+        """Check the user credentials against the web API.
+
+        Will raise evo.AuthenticationFailed if the credentials are invalid.
+        """
+
+        if (
+            self.client_v2 is None
+            or username != self.client_v2.username
+            or password != self.client_v2.password
+        ):
+            await self._load_auth_tokens(username)
+
+            client_v2 = evo.EvohomeClient(
+                username,
+                password,
+                **self._tokens,
+                session=self._session,
+            )
+
+        else:  # force a re-authentication
+            client_v2 = self.client_v2
+            client_v2._user_account = {}  # noqa: SLF001
```

Comment:
should you use clear() here perhaps?

Reply:
Hmmm... Looking at the library, this is better.
```python
client_v2._user_account = None  # noqa: SLF001
```